addons/point_of_sale/static/tests/unit/related_models_test.js

```diff
@@ -0,0 +1,948 @@
+            const p3 = models["product.product"].create({});
+            const t1 = models["product.tag"].create({ product_ids: [["link", p1, p2, p3]] });
+            const t2 = models["product.tag"].create({ product_ids: [["link", p1, p2]] });
+            // Test reading back the tags directly
+            const readT1 = models["product.tag"].read(t1.id);
+            assert.deepEqual(readT1, t1, "Tag 1 should be found and match the created tag");
+            const readP1 = models["product.product"].read(p1.id);
+            assert.deepEqual(readP1, p1, "Product should be found");
+            // Test the many2many relationship
+            assert.ok(
+                readT1.product_ids.includes(p1, p2, p3),
+                "Tag 1 should include Product 1, 2 and 3"
+            );
+            assert.ok(readP1.tag_ids.includes(t1, t2), "Product 1 should belong to Tag 1 and 2");
+            const readMany = models["product.product"].readMany([p2.id, p3.id]);
+            assert.deepEqual(readMany, [p2, p3], "Multiple products should be found");
+        });
+
+        QUnit.test("update operation, many2many", (assert) => {
+            const p1 = models["product.product"].create({});
+            const p2 = models["product.product"].create({});
+            const t1 = models["product.tag"].create({});
+            p1.update({ tag_ids: [["link", t1]] });
+            assert.ok(p1.tag_ids.includes(t1), "Product should be updated with new tag");
+            assert.ok(t1.product_ids.includes(p1), "Product should be linked to the new tag");
+            assert.notOk(
+                t1.product_ids.includes(p2),
+                "Product should be unlinked from the old tag"
+            );
+            t1.update({ product_ids: [["link", p2]] });
+            assert.ok(t1.product_ids.includes(p2), "Product should be linked to the tag");
+        });
+
+        QUnit.test("update operation, unlink many2many", (assert) => {
+            const p1 = models["product.product"].create({});
+            const t1 = models["product.tag"].create({});
+
+            t1.update({ product_ids: [["link", p1]] });
+            assert.ok(t1.product_ids.includes(p1), "Product should be linked to the tag");
+            assert.ok(p1.tag_ids.includes(t1), "Product should be linked to the tag");
+
+            t1.update({ product_ids: [["unlink", p1]] });
+            assert.notOk(t1.product_ids.includes(p1), "Product should be unlinked from the tag");
+            assert.notOk(p1.tag_ids.length > 0, "Product should be unlinked from the tag");
+            assert.ok(t1, "Product should be linked to the tag");
+        });
+
+        QUnit.test("update operation, Clear many2many", (assert) => {
+            const tag1 = models["product.tag"].create({});
+            const tag2 = models["product.tag"].create({});
+            const product = models["product.product"].create({ tag_ids: [[tag1, tag2]] });
+            product.update({ tag_ids: [["clear"]] });
```

Comment:
```suggestion
            assert.ok(product.tag_ids.length === 2, "tags are linked to the product");
            product.update({ tag_ids: [["clear"]] });
```

assert that the records are linked before update.